src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -0,0 +1,246 @@
+
+    final NodeRepository nodesRepository;
+    private final RootNodeInfoRepositoryProxy rootNodeInfoRepositoryProxy;
+
+    private static final String CATEGORY_BROKER_OUTPUT = NetworkModificationTreeService.class.getName() + ".output-broker-messages";
+
+    private static final Logger MESSAGE_OUTPUT_LOGGER = LoggerFactory.getLogger(CATEGORY_BROKER_OUTPUT);
+
+    @Autowired
+    private StreamBridge treeUpdatePublisher;
+
+    private void sendUpdateMessage(Message<String> message) {
+        MESSAGE_OUTPUT_LOGGER.debug("Sending message : {}", message);
+        treeUpdatePublisher.send("publishStudyUpdate-out-0", message);
+    }
+
+    private void emitNodeInserted(UUID studyUuid, UUID parentNode, UUID nodeCreated) {
+        sendUpdateMessage(MessageBuilder.withPayload("")
+            .setHeader(HEADER_STUDY_UUID, studyUuid)
+            .setHeader(HEADER_UPDATE_TYPE, NODE_CREATED)
+            .setHeader(HEADER_PARENT_NODE, parentNode)
+            .setHeader(HEADER_NEW_NODE, nodeCreated)
+            .build()
+        );
+    }
+
+    private void emitNodesChanged(UUID studyUuid, Collection<UUID> nodes) {
+        sendUpdateMessage(MessageBuilder.withPayload("")
+            .setHeader(HEADER_STUDY_UUID, studyUuid)
+            .setHeader(HEADER_UPDATE_TYPE, NODES_UPDATED)
+            .setHeader(HEADER_NODES, nodes)
+            .build()
+        );
+    }
+
+    private void emitNodesDeleted(UUID studyUuid, Collection<UUID> nodes, boolean deleteChildren) {
+        sendUpdateMessage(MessageBuilder.withPayload("")
+            .setHeader(HEADER_STUDY_UUID, studyUuid)
+            .setHeader(HEADER_UPDATE_TYPE, NODES_DELETED)
+            .setHeader(HEADER_NODES, nodes)
+            .setHeader(HEADER_REMOVE_CHILDREN, deleteChildren)
+            .build()
+        );
+    }
+
+    @Autowired
+    public NetworkModificationTreeService(NodeRepository nodesRepository,
+                                          RootNodeInfoRepository rootNodeInfoRepository,
+                                          ModelNodeInfoRepository modelNodeInfoRepository,
+                                          NetworkModificationNodeInfoRepository networkModificationNodeInfoRepository
+    ) {
+        this.nodesRepository = nodesRepository;
+        this.rootNodeInfoRepositoryProxy = new RootNodeInfoRepositoryProxy(rootNodeInfoRepository);
+        repositories.put(NodeType.ROOT, rootNodeInfoRepositoryProxy);
+        repositories.put(NodeType.MODEL, new ModelNodeInfoRepositoryProxy(modelNodeInfoRepository));
+        repositories.put(NodeType.NETWORK_MODIFICATION, new NetworkModificationNodeInfoRepositoryProxy(networkModificationNodeInfoRepository));
+
+    }
+
+    @Transactional
+    public Mono<AbstractNode> createNode(UUID id, AbstractNode nodeInfo) {
+        return Mono.fromCallable(() -> {
+            Optional<NodeEntity> parentOpt = nodesRepository.findById(id);
```

Comment:
I think this code is executed outside of the transaction ?

Reply:
in other services we moved all the database access to a separate doXXX method which is transactional, so that when this method returns and webflux subscribes and executes the mono, it starts the transaction and accesses the database.